src/AbstractTypedContainerImplementation.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\ServiceManager;
+
+use Psr\Container\ContainerInterface;
+
+use function assert;
+
+/**
+ * @psalm-require-extends ServiceManager
+ */
+abstract class AbstractTypedContainerImplementation implements ContainerInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function has(string $id): bool
+    {
+        assert($this instanceof ServiceManager);
+        /** @psalm-suppress InaccessibleMethod We actually can access the method, but this is hacky code. */
+        return $this->hasService($id);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function get(string $id)
```

Comment:
should we also add `mixed` as return type here as it will be added with `psr/container` v3 anyways?

Reply:
Yes, please. It's new code; let's be explicit.

Reply:
Ah, can't really add return type `:mixed` since `psr/container` v2 has minimum of PHP 7.4 where `mixed` as a return value is not allowed...